### PR TITLE
feature: trace sandbox creation

### PR DESCRIPTION
**Pull request by rgrinberg**
Signed-off-by: Rudi Grinberg <me@rgrinberg.com>

ps-id: 50198f48-67bb-4f8b-8e2c-daedd81973a6

### CODE REVIEW

File: src/dune_engine/sandbox.ml [hunk 1]
```diff
@@ -116,6 +135,22 @@ let create ~mode ~rule_loc ~deps ~rule_dir ~rule_digest ~expand_aliases =
   (* CR-someday amokhov: Note that this doesn't link dynamic dependencies, so
      targets produced dynamically will be unavailable. *)
   link_deps t ~mode ~deps;
+  let () =
+    match (common_metrics, dune_stats) with
+    | Some _, None | None, Some _ -> assert false
+    | None, None -> ()
+    | Some common, Some dune_stats ->
+      let now = Unix.gettimeofday () in
```
Comment (snowleopard) [inline src/dune_engine/sandbox.ml:143]: All this messing with times looks horrible. Can't we move it out to some helper module, not to pollute the sandboxing logic?

Comment (snowleopard) [reply src/dune_engine/sandbox.ml:143]: (I'd expect the diff in this module to be 2-3 lines, not 37!)

Comment (rgrinberg) [reply src/dune_engine/sandbox.ml]: Couldn't reduce to 2-3, but hopefully this is nimble enough.

File: src/dune_engine/sandbox.ml [hunk 2]
```diff
@@ -102,7 +102,15 @@ let snapshot t =
   in
   walk (Path.build t.dir) Path.Map.empty
 
-let create ~mode ~rule_loc ~deps ~rule_dir ~rule_digest ~expand_aliases =
+let create ~mode ~dune_stats ~rule_loc ~deps ~rule_dir ~rule_digest
+    ~expand_aliases =
+  let event =
+    Dune_stats.start dune_stats (fun () ->
+        let cat = Some [ "sandbox" ] in
```
Comment (snowleopard) [inline src/dune_engine/sandbox.ml:109]:
```suggestion
        let cat = Some [ "create-sandbox" ] in
```
^ A bit more precise I guess.

File: src/dune_engine/sandbox.ml (hunk 1 above)
Comment (snowleopard) [reply src/dune_engine/sandbox.ml]: Thanks, that's much better!

**Review by snowleopard — APPROVED** [review]
Thanks!

**Pull request merged**